_posts/2023-10-23-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,257 @@
+If you exceed the configured max queue size, the UTO will print an error and then shutdown (Kubernetes will take care of restarting the pod).
+In that case, you can simply raise the max queues size to avoid the periodic operator restarts.
+
+### Race condition
+
+There is a known race condition happening when the operator and external applications try to change a topic configuration concurrently.
+Given the way Kafka is implemented, there is no solution to that problem, but the UTO significantly reduces the likelihood of reconciliation failures.
+
+When the external application wins, the topics will be created using the default Kafka cluster configuration.
+In this case, the BTO simply fails the reconciliation, and you need to recover manually.
+Instead, the UTO always reconciles, which results in reconfiguration only when the default configuration differs from the spec.
+That reconfiguration will either succeed, or fail.
+
+The recommendation here is to configure the Kafka cluster with `auto.create.topics.enable: false` in order to reduce the likelihood of hitting this problem.
+Unfortunately, disabling topic creation doesn't help with applications that use the Kafka Admin client to directly create required topics (i.e. Kafka Streams based applications).
+In this case, the application should be written to wait for topic existence, or use an init container to do the same.
+
+### Finalizers
+
+[Finalizers](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers) are used by default to avoid missing topic deletion events when the UTO is not running.
+A common pitfall is that the namespace becomes stuck in a "terminating" state when you try to delete it without first removing all finalizers.
+If this happens, you can simply remove finalizers from all `KafkaTopic` resources at once with the following command.
+
+```sh
+kubectl get kt -o yaml | yq 'del(.items[].metadata.finalizers[])' | kubectl apply -f -
+```
+
+It is also possible to disable finalizers using the following configuration.
+
+```sh
+# from Kafka custom resource
+spec:
+  entityOperator:
+    userOperator: {}
+    topicOperator: {}
+    template:
+      topicOperatorContainer:
+        env:
+          - name: STRIMZI_USE_FINALIZERS
+            value: "false"
+
+# from standalone Deployment
+spec:
+  template:
+    spec:
+      containers:
+        - name: STRIMZI_USE_FINALIZERS
+          value: "false"
+```
+
+### Upgrading from BTO to UTO
+
+The UTO is available as alpha feature since Strimzi 0.36.0, so we need to enable a feature gate in order to use it.
+[Strimzi 0.38.0](https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.38.0) adds Prometheus metrics and pause reconciliation feature.
+
+**This is not production ready yet, but the plan is to move it to beta in Strimzi 0.39.0, where it will be enabled by default.**
+**The UTO might require different CPU and memory resources, so we recommend to monitor and adjust accordingly.**
```

Comment:
I would either add specific (e.g. double the memory) or leave it out to be honest.